src/backend/dx11/shaders/clear.hlsl

```diff
@@ -0,0 +1,22 @@
+cbuffer ClearColorF32   : register(b0) { float4 ClearF32; };
+cbuffer ClearColorU32   : register(b0) { uint4 ClearU32; };
+cbuffer ClearColorI32   : register(b0) { int4 ClearI32; };
+cbuffer ClearColorDepth : register(b0) { float ClearDepth; };
+
+// fullscreen triangle
+float4 vs_partial_clear(uint id : SV_VertexID) : SV_Position
+{
+    return float4(
+        float(id / 2) * 4.0 - 1.0,
+        float(id % 2) * 4.0 - 1.0,
+        0.0,
+        1.0
+    );
+}
+
+// TODO: send constants through VS as flat attributes
```

Comment:
I don't think this is going to be better. Constants are fine, and I doubt we are ALU/register limited here anyway ;)